src/index.jsx

```diff
@@ -0,0 +1,76 @@
+import React, { useState } from 'react'
+import ReactDOM from 'react-dom'
+
+
+function Habit({ onDelete, habit, keyValue }) {
+  return (
+    <div>
+      {habit}
+      <button onClick={() => onDelete(keyValue)} type="button" >완료</button>
+    </div>
+  )
+}
+
+function Habits({ onDelete, habits }) {
+  return (
+    <div>
+      {
+        habits.map((habit, key) => (
+          <Habit onDelete={onDelete} habit={habit} key={key} keyValue={key} />
+        ))
+      }
+    </div>
+  )
+}
+
+function Input({ onSubmit, onDelete, habits }) {
+
+  return (
+    <div>
+      <div>
+        <form onSubmit={onSubmit}>
+          <input type="text" name="content" placeholder="할 일을 입력해 주세요" />
+          <input type="submit" value="추가" />
+        </form>
+      </div>
+      <div>
+        {habits.length === 0 ? "할 일이 없어요!" : <Habits onDelete={onDelete} habits={habits} />}
+      </div>
+    </div>
+  )
+}
+
+function Page({ onSubmit, onDelete, habits }) {
+  return (
+    <Input onSubmit={onSubmit} onDelete={onDelete} habits={habits} />
+  )
+}
+
+function App() {
+  const [state, setState] = useState({
+    habits: []
+  })
+  const { habits } = state;
+
+  function handleDelete(key) {
+    setState({
+      habits: habits.filter((value, index) => index != key)
+    }
+    )
+  }
+
+  function handleSubmit(e) {
+    e.preventDefault()
+    setState({
+      habits: [e.target[0].value, ...habits]
+    })
```

Comment:
```suggestion
    setState({
      ...state
      habits: [e.target[0].value, ...habits]
    });
```

만약 객체에 habits 말고 다른 상태 속성이 있으면 어떤 일이 발생할까요?

Reply:
그럼 habits만 남겠네요.. 리뷰 감사합니다!!